src/223/main/kotlin/org/rust/ide/hints/codeVision/RsImplementationsCodeVisionProvider.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints.codeVision
+
+import com.intellij.codeInsight.CodeInsightBundle
+import com.intellij.codeInsight.codeVision.CodeVisionRelativeOrdering
+import com.intellij.codeInsight.daemon.impl.PsiElementListNavigator
+import com.intellij.codeInsight.hints.codeVision.InheritorsCodeVisionProvider
+import com.intellij.ide.util.DefaultPsiElementCellRenderer
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.NavigatablePsiElement
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.CachedValue
+import com.intellij.psi.util.CachedValueProvider
+import com.intellij.psi.util.CachedValuesManager
+import com.intellij.psi.util.PsiModificationTracker
+import com.intellij.util.EmptyQuery
+import com.intellij.util.Query
+import org.rust.RsBundle
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsTraitItem
+import org.rust.lang.core.psi.ext.*
+import org.rust.openapiext.isUnitTestMode
+import java.awt.event.MouseEvent
+
+class RsImplementationsCodeVisionProvider : InheritorsCodeVisionProvider() {
+    override fun acceptsFile(file: PsiFile): Boolean = file is RsFile
+
+    override fun acceptsElement(element: PsiElement): Boolean {
+        if (!isUnitTestMode && !Registry.`is`("org.rust.code.vision.implementation", false)) return false
+
+        return when (element) {
+            is RsTraitItem -> true
+            is RsAbstractable -> element.owner is RsAbstractableOwner.Trait
+            else -> false
+        }
+    }
+
+    override fun getHint(element: PsiElement, file: PsiFile): String? {
+        val rsElement = element as? RsElement ?: return null
+        if (rsElement !is RsTraitItem && rsElement !is RsAbstractable) return null
```

Comment:
nit: if replace `val rsElement = element as? RsElement ?: return null` with `if (element !is RsElement) return null`, you will be able to use `element` and not to introduce a new variable with a new name

But in this case, you even don't need `val rsElement = element as? RsElement ?: return null` at all since you have more strict check at the next line.

Reply:
The following check doesn't help, because Kotlin isn't smart enough to smart cast to RsElement after checking that it isn't anstractable or trait (and there is no unifying interface for abstractables and trait I think).

Reply:
Oh, I see. But you still can avoid introducing redundant `rsElement` variable